src/tiled/scriptmodule.cpp

```diff
@@ -651,6 +651,10 @@ void ScriptModule::currentDocumentChanged(Document *document)
     emit activeAssetChanged(document ? document->editable() : nullptr);
 }
 
+EditableProject *ScriptModule::project() const
+{
+    return new EditableProject(&ProjectManager::instance()->project(), MainWindow::maybeInstance());
```

Comment:
Why is the parent of the `EditableProject` the `MainWindow`? I think that will mean that these instances are never deleted, and essentially leaking every time the `project` property is accessed.

I think we have two options:

* We instantiate one `EditableProject` in the `ScriptModule`, owned by the `ScriptModule`. We can do this on-demand, remember the instance in a member variable so that we can return the same instance the next time.
* We set ownership to the JS engine, in which case it can be cleaned up by the garbage collector.

I think I'd prefer the first option in this case, because we only need one instance. The `Project*` is never going to be different after all, since it always points to the `Project` member of the `ProjectModel`.

Reply:
Just my misunderstanding. I'll try your first option.